engine/src/main/scala/cromwell/webservice/MetadataBuilderActor.scala

```diff
@@ -34,7 +34,7 @@ object MetadataBuilderActor {
   private val KeySeparator = ':'
   private val ListNameCaptureID = "listName"
   private val ListIndexCaptureID = "listIndex"
-  private val ListMatcher = "^(.+)\\[([a-zA-Z0-9]+)\\]$".r(ListNameCaptureID, ListIndexCaptureID)
+  private val ListMatcher = "^(.+)\\[([^\\]\\[]+)\\]$".r(ListNameCaptureID, ListIndexCaptureID)
```

Comment:
Sorry I was slow with an edit. Yes, they're ints so we just need to make them positive. We'll increase the chance of a collision by a factor of sqrt(2) but... (shrug)